src/dataStructures/linkedList/LinkedList.java

```diff
@@ -103,9 +85,9 @@ public boolean insert(T object, int idx) {
     for (int i = 0; i < idx; i++) {
       prev = trav;
       trav = trav.next;
-    }
+    } // This iteration allows us to store a copy of nodes before and after idx.
```

Comment:
this line can be slightly confusing, i think omitting would be better. People reading the code are expected to have some foundation in cs and data structures anyway so it should be gud!